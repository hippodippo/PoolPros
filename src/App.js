import React, { Component } from 'react';
import './styles/normalize.scss';
import './styles/App.scss';
import Nav from './components/Nav/Nav';
import routes from './routes';
// import Footer from './components/Footer/Footer';
import Wrapper from './components/Wrapper/Wrapper';

class App extends Component {
  constructor(props, context) {
    super(props, context);

    // Set initial state.
    this.state = {
      visible: false,
    };

    this.toggleMenu = this.toggleMenu.bind(this);
    this.handleMouseDown = this.handleMouseDown.bind(this);
  }

  // Toggles Mobile Menu.
  toggleMenu() {
    this.setState({
      visible: !this.state.visible,
    });
  }

  handleMouseDown(e) {
    this.toggleMenu();

    e.stopPropagation();
  }

  render() {
    return (
      <div className="App">
        <Nav handleMouseDown={this.handleMouseDown} menuVisibility={this.state.visible} />
        <Wrapper>
          { routes }
        </Wrapper>
        {/* <Footer /> */}
      </div>
    );
  }
}

export default App;