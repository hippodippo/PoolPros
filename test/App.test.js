import React from 'react';
import { shallow } from 'enzyme';
import App from '../src/App.js';

describe('App', () => {
  test('Should match snapchat', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find('h1').text()).toBe('Hello Kaycee!'); // Placeholder Test.
    expect(wrapper).toMatchSnapshot();
  });
});