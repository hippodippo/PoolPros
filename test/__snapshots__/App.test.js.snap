// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapchat 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello Kaycee!",
    },
    "ref": null,
    "rendered": "Hello Kaycee!",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello Kaycee!",
      },
      "ref": null,
      "rendered": "Hello Kaycee!",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
